RXPromise Libraries/Source/RXPromise.mm

```diff
@@ -357,10 +353,8 @@ - (void)synced_updateWithProgress:(float)progress {
     if (_state != Pending) {
         return;
     }
-    OSSpinLockLock(&_progressSpinLock);
-    _progressValue = progress;
-    if (_progressHandlers) {
-        for (const std::pair<id, promise_progressHandler_t> handlerPair : *_progressHandlers) {
+    if (_progressHandlers.size()) {
```

Comment:
`if (_progressHandlers.size())` can be omitted. Just iterating over an empty list should work, too.